File: backend/app/models/abandoned_cart.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from prisma.enums import CartStatus

class CartItemSchema(BaseModel):
    id: int
    name: str
    quantity: int
    price: float
    image: Optional[str] = None

    class Config:
        from_attributes = True  # Allows parsing Prisma items

class UserSchema(BaseModel):
    id: int
    first_name: str
    last_name: str
    email: str

    class Config:
        from_attributes = True

class AbandonedCartSchema(BaseModel):
    id: int
    user: Optional[UserSchema]
    email: Optional[str]
    cart_number: str
    status: Optional[CartStatus]
    items: Optional[list[CartItemSchema]] = []
    total: float = 0
    subtotal: float = 0
    tax: float = 0
    shipping_fee: float = 0
    created_at: datetime

    class Config:
        from_attributes = True

class PaginatedAbandonedCarts(BaseModel):
    items: List[AbandonedCartSchema]
    next_cursor: Optional[int] = None
    limit: int

File: app/src/schemas/abandoned-cart.ts
export interface UserMini {
    first_name: string;
    last_name: string;
    phone: string | null;
    email: string | null;
}

export interface CartItemVariant {
    id: number;
    sku: string;
    price: number;
    compare_at_price: number | null;
    options: Record<string, string> | null;
}

export interface CartItem {
    id: number;
    variant_id: number;
    name: string | null;
    slug: string | null;
    image: string | null;
    quantity: number;
    variant: CartItemVariant | null;
}

export interface AbandonedCart {
    id: number;
    cart_number: string;
    user_id: number | null;
    user: UserMini | null;
    guest_id: string | null;
    status: string;
    email: string | null;
    phone: string | null;
    subtotal: number;
    shipping_fee: number;
    discount_amount: number;
    total: number;
    coupon_code: string | null;
    items: CartItem[];
    created_at: string;
    updated_at: string | null;
}

export interface AbandonedCartListResponse {
    items: AbandonedCart[];
    total: number;
    skip: number;
    limit: number;
    has_more: boolean;
}

export interface AbandonedCartListParams {
    skip?: number;
    limit?: number;
    search?: string;
}
